src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_ReturnKey.cs

```diff
@@ -62,9 +62,13 @@ private void CommitOnEnter(out bool sendThrough, out bool committed)
 
             if (model.IsSoftSelection)
             {
-                // If the completion list is soft selected, then don't commit on enter.  Instead,
-                // send the enter through to the editor and dismiss the completion list.
-                sendThrough = true;
+                // If the completion list is soft selected, then don't commit on enter.
+                // Instead, just dismiss the completion list.  If we're in a normal editor
+                // or the Immediate window, we'll also send the enter through to the editor.
+                // In single-line debugger windows (Watch, etc), however, we don't want to
+                // send the enter though, because those windows don't support displaying
+                // more than one line of text.
+                sendThrough = !model.TriggerInfo.IsDebugger || model.TriggerInfo.IsImmediateWindow;
                 committed = false;
                 return;
```

Comment:
C# IntelliSense has the option "send enter to editor after fully typed word" (something like that). If the user has the option set and fully types a hard-selected item and then hits enter, we'll send it through and they'll see a blank watch window. Might be simpler to do the check in the caller of CommitOnEnter, when we look at "committed" to see if we should call nextHandler().